### PR TITLE
GAP-2537:add logic to handle application with no adverts

### PR DESCRIPTION
**Pull request by ConnorTCO**
## Description

Ticket # and link

Summary of the changes and the related issue. List any dependencies that are required for this change:

## Type of change

Please check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes:

- [ ] Unit Test

- [ ] Integration Test (if applicable)

- [ ] End to End Test (if applicable)

## Screenshots (if appropriate):

Please attach screenshots of the change if it is a UI change:

# Checklist:

- [ ] If I have listed dependencies above, I have ensured that they are present in the target branch.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation where applicable.
- [ ] I have ran cypress tests and they all pass locally.

### CODE REVIEW

File: src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java [hunk 1]
```diff
@@ -67,36 +69,52 @@ public ResponseEntity<GetGrantSchemeWithApplicationAndAdverts> getGrantSchemeByI
...
+        checkIfAdvertExistAndHasApplyingUrlPointingToInternalApplication(grantSchemeId, dto);
 
         log.debug("Scheme with ID {} is for internal application: {} and has a published application form : {}",
                 grantSchemeId, dto.isHasInternalApplication(), dto.isHasPublishedInternalApplication());
 
         return ResponseEntity.ok(dto);
     }
 
+
+    private void checkIfAdvertExistAndHasApplyingUrlPointingToInternalApplication(Integer grantSchemeId, SchemeMandatoryQuestionApplicationFormInfosDto dto) {
```
Comment (jgunnCO) [inline src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java:97]:
```suggestion
    private void checkIfAdvertExistAndIsInternal(Integer grantSchemeId, SchemeMandatoryQuestionApplicationFormInfosDto dto) {
```

File: src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java [hunk 2]
```diff
@@ -67,36 +69,52 @@ public ResponseEntity<GetGrantSchemeWithApplicationAndAdverts> getGrantSchemeByI
...
 
-        log.debug("Getting Advert associated to scheme with id {}", grantSchemeId);
-        final GrantAdvert advert = grantAdvertService.getAdvertBySchemeId(grantSchemeId.toString());
-        log.debug("Advert with id {} found for scheme with id {}", advert.getId(), grantSchemeId);
-
-        final String webpageUrl = grantAdvertService.getExternalSubmissionUrl(advert);
-
-        final boolean webPageUrlIsForInternalApplications = webpageUrl.contains(environmentProperties.getFrontEndUri());
-        log.debug("Advert is pointing to an internal application form : {}", webPageUrlIsForInternalApplications);
+        final boolean hasSchemeAnApplication = grantApplicationService.doesSchemeHaveAnApplication(grantScheme);
```
Comment (ConorFayleTCO) [inline src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java:78]:
```suggestion
        final boolean schemeHasAnApplication = grantApplicationService.doesSchemeHaveAnApplication(grantScheme);
```

File: src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java [hunk 3]
```diff
@@ -67,36 +69,52 @@ public ResponseEntity<GetGrantSchemeWithApplicationAndAdverts> getGrantSchemeByI
...
+        final boolean hasPublishedApplication = grantApplicationService.doesSchemeHaveAPublishedApplication(grantScheme);
+        log.debug("Application form associated to scheme with Id {} has PUBLISHED status: {}", grantScheme.getId(), hasPublishedApplication);
 
         final SchemeMandatoryQuestionApplicationFormInfosDto dto = SchemeMandatoryQuestionApplicationFormInfosDto.builder()
-                .hasInternalApplication(webPageUrlIsForInternalApplications)
-                .hasPublishedInternalApplication(false)
+                .hasAdvertPublished(false)
+                .hasInternalApplication(hasSchemeAnApplication)
```
Comment (ConorFayleTCO) [inline src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java:84]:
```suggestion
                .hasInternalApplication(schemeHasAnApplication)
```

File: src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java [hunk 4]
```diff
@@ -67,36 +69,52 @@ public ResponseEntity<GetGrantSchemeWithApplicationAndAdverts> getGrantSchemeByI
...
+                .hasPublishedInternalApplication(hasPublishedApplication)
                 .build();
 
-        if (webPageUrlIsForInternalApplications) {
-            final boolean hasPublishedApplication = grantApplicationService.doesSchemeHaveAPublishedApplication(grantScheme);
-            log.debug("Application form associated to advert with Id {} has PUBLISHED status: {}", advert.getId(), hasPublishedApplication);
-            dto.setHasPublishedInternalApplication(hasPublishedApplication);
-        }
+        checkIfAdvertExistAndHasApplyingUrlPointingToInternalApplication(grantSchemeId, dto);
```
Comment (ConorFayleTCO) [inline src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java:88]: as per Jamie's comment
```suggestion
        checkIfAdvertExistAndIsInternal(grantSchemeId, dto);
```

File: src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java (hunk 2 above)
Comment (tco-olivergodby) [reply src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java:78]: I agree with Connor's suggestion here, FWIW

File: src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java (hunk 3 above)
Comment (tco-olivergodby) [reply src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java:84]: Same here, let's adopt Conor's suggestion.

**Review by tco-olivergodby — APPROVED** [review]
This is good to go - merge when ready

**Pull request merged**